src/Service/MigrationService.php

```diff
@@ -0,0 +1,571 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Yii\Db\Migration\Service;
+
+use Yiisoft\Arrays\ArrayHelper;
+use Yiisoft\Db\Connection\Connection;
+use Yiisoft\Db\Exception\Exception;
+use Yiisoft\Db\Exception\InvalidConfigException;
+use Yiisoft\Db\Exception\NotSupportedException;
+use Yiisoft\Db\Query\Query;
+use Yiisoft\Yii\Console\ExitCode;
+use Yiisoft\Yii\Db\Migration\MigrationInterface;
+use Yiisoft\Yii\Db\Migration\Helper\ConsoleHelper;
+
+use function array_slice;
+use function array_values;
+use function closedir;
+use function file_exists;
+use function gmdate;
+use function is_file;
+use function ksort;
+use function opendir;
+use function preg_match;
+use function readdir;
+use function str_replace;
+use function strcasecmp;
+use function strpos;
+use function time;
+use function trim;
+use function usort;
+
+final class MigrationService
+{
+    /**
+     * The name of the dummy migration that marks the beginning of the whole migration history.
+     */
+    public const BASE_MIGRATION = 'm000000_000000_base';
+
+    private string $createNamespace = '';
+    private string $createPath = '';
+    private array $updateNamespace = [];
+    private array $updatePath = [];
+    private string $comment = '';
+    private bool $compact = false;
+    private array $fields = [];
+    private array $generatorTemplateFiles = [
+        'create' => '@views/migration.php',
+        'table' => '@views/createTableMigration.php',
+        'dropTable' => '@views/dropTableMigration.php',
+        'addColumn' => '@views/addColumnMigration.php',
+        'dropColumn' => '@views/dropColumnMigration.php',
+        'junction' => '@views/createTableMigration.php',
+    ];
+    private int $maxNameLength = 180;
+    private int $migrationNameLimit = 0;
+    private string $migrationTable = '{{%migration}}';
+    private string $templateFile = '@views/migration.php';
```

Comment:
Either should be able to customize it or it should be made local variable instead of class property.

Reply:
done.

Reply:
https://github.com/terabytesoftw/yii-db-migration/blob/02eabab4b74d252d15638cac1519c05a81edf663/src/Service/MigrationService.php#L375